eth/protocols/snap/sync.go

```diff
@@ -1321,6 +1352,10 @@ func (s *Syncer) assignTrienodeHealTasks(success chan *trienodeHealResponse, fai
 		if cap > maxTrieRequestCount {
 			cap = maxTrieRequestCount
 		}
+		cap = int(float64(cap) / s.trienodeHealThrottle)
+		if cap <= 0 {
+			cap = 1
+		}
```

Comment:
Not sure I understand this. Afaiu, this section is where we have an idle peer, and we want to assign requests to that peer. This `cap` is the max that we want to assign to a single peer, previously `1024` . 
When you now modify this, it means that we will throttle the amount we fetch from each peer. So if we have 200 peers, we will make 200 requests, each requesting a single trie node. 
Instead making one single request to the best peer, for those 200 trienodes.

Reply:
Yes, I need to limit the amount of data I request globally. I could ask all of it from a single peer - and wait for the response - or I can ask for small batches spread out across many peers and have a stabler stream of inbound data + less load on anyone specific.